### PR TITLE
Merge master into SeedData_Xiahui

### PR DESCRIPTION
**Pull request by Xia-Hui-Liu**

### CODE REVIEW

**Review by jmc317 — APPROVED** [review]

File: Models/Entities/ParkedVehicle.cs
Comment (Kasleets) [inline Models/Entities/ParkedVehicle.cs:1]: What's the reason behind removing the LINQ expression on ArrivalTime to the block version?

How do you get the Vehicle type if it is defined as VehicleType VehicleType?

File: Models/Entities/VehicleType.cs
Comment (Kasleets) [inline Models/Entities/VehicleType.cs:1]: I do understand the idea behind removing Boat and airplane, need to do it properly though, the last property shouldn't have the ' , '

**Review by Kasleets — CHANGES_REQUESTED** [review]
Overall looks good. I would request adding comments while coding, because it makes it harder to read if there is none.

Please answer the two annotations that I submitted inside the review.

File: Models/Entities/ParkedVehicle.cs
Comment (Xia-Hui-Liu) [reply Models/Entities/ParkedVehicle.cs:1]: 1. The reason I used a constructor to set the ArrivalTime property in  ParkedVehicle class is to initialization it as soon as an instance of ParkedVehicle is created. I don't know what's the benifit to use LINQ expression.
2. Set the folder as global using in Views => shared =>_ViewImports.cshtml(@using Garage2._0.Models.Entities), then use it in ApplicationDbContext(using Garage2._0.Models.Entities;)

File: Models/Entities/VehicleType.cs
Comment (Xia-Hui-Liu) [reply Models/Entities/VehicleType.cs:1]: sure

File: Models/Entities/ParkedVehicle.cs
Comment (Kasleets) [reply Models/Entities/ParkedVehicle.cs:1]: Benefit of using LINQ is the code clarity and minimalizing the code bulk.

**Review by Kasleets — COMMENTED** [review]
Should be alright to merge to master branch, let's wait until 15:00 meeting to review it together as a team, because we have overlapping two code bases, to avoid breaking the system.

**Pull request closed without merging**